packages/backend/src/auth-service-client/client.ts

```diff
@@ -0,0 +1,90 @@
+interface Tenant {
+  id: string
+  idpConsentUrl: string
+  idpSecret: string
+}
+
+export class AuthServiceClientError extends Error {
+  constructor(
+    message: string,
+    public status: number,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public details?: any
+  ) {
+    super(message)
+    this.status = status
+    this.details = details
+  }
+}
+
+export class AuthServiceClient {
+  private baseUrl: string
+
+  constructor(baseUrl: string) {
+    this.baseUrl = baseUrl
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private async request<T>(path: string, options: RequestInit): Promise<T> {
+    const response = await fetch(`${this.baseUrl}${path}`, options)
+
+    if (!response.ok) {
+      let errorDetails
+      try {
+        errorDetails = await response.json()
+      } catch {
+        errorDetails = { message: response.statusText }
+      }
+
+      throw new AuthServiceClientError(
+        `Auth Service Client Error: ${response.status} ${response.statusText}`,
+        response.status,
+        errorDetails
+      )
+    }
+
+    if (
+      response.status === 204 ||
+      response.headers.get('Content-Length') === '0'
+    ) {
+      return undefined as T
+    }
+
+    const contentType = response.headers.get('Content-Type')
+    if (contentType && contentType.includes('application/json')) {
+      try {
+        return (await response.json()) as T
+      } catch (error) {
+        throw new AuthServiceClientError(
+          `Failed to parse JSON response from ${path}`,
+          response.status
+        )
+      }
+    }
+
+    return (await response.text()) as T
+  }
+
+  public tenant = {
+    get: (id: string) =>
+      this.request<Tenant>(`/tenant/${id}`, { method: 'GET' }),
```

Comment:
do we need the headers object in here as well? 
Can also just have it in the `request` fn

Reply:
The second arg in request? Yeah I think we need this. Its more than just the headers. `request` just wraps `fetch` and fetch needs the the `method` at the very least since its used for gets, posts, delete, etc.

Reply:
As in, I think we need `{ 'Content-Type': 'application/json' }` also for the GET method.

My suggestion is to just have `{ 'Content-Type': 'application/json' }` once on L29

Reply:
i added it in `request` as suggested. Actually I dont think we need it for the `GET` (just for specifying format of body no?) but it shouldn't hurt in any case.